allensdk/test/brain_observatory/behavior/behavior_project_cache_data_model/test_behavior_project_cache.py

```diff
@@ -0,0 +1,250 @@
+    expected = expected_behavior_session_table['df']
+    if cache.cache:
+        path = cache.manifest.path_info.get("behavior_sessions").get("spec")
+        assert os.path.exists(path)
+
+    safe_df_comparison(expected, obtained)
+
+
+@pytest.mark.parametrize("TempdirBehaviorCache, "
+                         "expected_experiments_table",
+                         [(True, True),
+                          (True, False),
+                          (False, True),
+                          (False, False)], indirect=True)
+def test_get_experiments_table(TempdirBehaviorCache,
+                               expected_experiments_table):
+    cache = TempdirBehaviorCache
+    obtained = cache.get_ophys_experiment_table(
+                     passed_only=expected_experiments_table['passed_only'])
+    if cache.cache:
+        path = cache.manifest.path_info.get("ophys_experiments").get("spec")
+        assert os.path.exists(path)
+
+    safe_df_comparison(expected_experiments_table['df'], obtained)
+
+
+@pytest.mark.parametrize("TempdirBehaviorCache", [True], indirect=True)
+def test_session_table_reads_from_cache(TempdirBehaviorCache,
+                                        caplog):
+    caplog.set_level(logging.INFO, logger="call_caching")
+    cache = TempdirBehaviorCache
+    cache.get_ophys_session_table()
+    expected_first = [
+        ('call_caching', logging.INFO, 'Reading data from cache'),
+        ('call_caching', logging.INFO, 'No cache file found.'),
+        ('call_caching', logging.INFO, 'Fetching data from remote'),
+        ('call_caching', logging.INFO, 'Writing data to cache'),
+        ('call_caching', logging.INFO, 'Reading data from cache'),
+        ('call_caching', logging.INFO, 'Reading data from cache'),
+        ('call_caching', logging.INFO, 'No cache file found.'),
+        ('call_caching', logging.INFO, 'Fetching data from remote'),
+        ('call_caching', logging.INFO, 'Writing data to cache'),
+        ('call_caching', logging.INFO, 'Reading data from cache'),
+        ('call_caching', logging.INFO, 'Reading data from cache'),
+        ('call_caching', logging.INFO, 'No cache file found.'),
+        ('call_caching', logging.INFO, 'Fetching data from remote'),
+        ('call_caching', logging.INFO, 'Writing data to cache'),
+        ('call_caching', logging.INFO, 'Reading data from cache'),
+        ('call_caching', logging.INFO, 'Reading data from cache')]
+    assert expected_first == caplog.record_tuples
+    caplog.clear()
+    cache.get_ophys_session_table()
+    assert [expected_first[0]]*4 == caplog.record_tuples
+
+
+@pytest.mark.parametrize("TempdirBehaviorCache", [True], indirect=True)
+def test_behavior_table_reads_from_cache(TempdirBehaviorCache,
+                                         caplog):
+    caplog.set_level(logging.INFO, logger="call_caching")
+    cache = TempdirBehaviorCache
+    cache.get_behavior_session_table()
+    expected_first = [
```

Comment:
this method of checking that something happened seems indirect, and prone to failure if someone changed logging messages,
I think this is a good use case for [assert_called_once](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once) if you can modify the test to work with `unittest.mock`

Reply:
Yeah. In fact, that test is now failing on-prem because there's an unexpected logger message about the number of threads that NumExpr is using.

I was going to just change it to assert that each of the distinct logging messages was in caplog (rather than verifying the detailed contents of caplog). Does that seem like a reasonable solution?

(The test dates back to fa93ebd, apparently)